Misc. improvements to output diagnostics

bendudson opened the pull request.

bendudson committed:
anomalous_diffusion: Always save diffusion coefficients

The coefficients are not time dependent, so always save as time
independent fields, since this does not take much storage space.

bendudson committed:
evolve_density: fix flow_ylow initialisation

If parallel flow is not present, then flow_ylow would not be initialised.
This could then cause an operation on an empty field later.

bendudson committed:
evolve_density/pressure/energy: Save time independent sources

Only add a time_dimension attribute if the source is time dependent.
This saves storage space, and makes clear in the output whether the
source was time dependent.

bendudson committed:
vorticity: Adding transfer channel diagnostics

Energy exchange between ExB flow and other forms. Hopefully
helpful in understanding the flow of energy through the system.

mikekryjak commented:
This is great, I forgot that post-processing variables will be saved just once if you don't include the time dimension. This should go into https://github.com/boutproject/hermes-3/issues/299.

bendudson committed:
evolve_pressure/momentum: Add energy diagnostics

If `diagnose = true`, save energy transfer and advection channels.

bendudson committed:
polarisation_drift: Only save phi_pol if advection enabled

phi_pol is only calculated if polarisation drift advection is enabled.

bendudson committed:
fix new diagnostics

- Can't have "/" in the variable name
- Use getNonFinal when retrieving the value so the value can be modified.
